### PR TITLE
GH-7335: Fix space key interaction for clickable Datatable rows

### LINKED ISSUE #7335: DataTable - clickable rows keyboard selection
### Expected Behavior
Row in a Datatable that are clickable should be interactive via both 'space' and 'enter'

### Actual Behavior
Trying to interact with the row with 'space' doesn't work, pressing 'enter' sometimes works

### URL, screen shot, or Codepen exhibiting the issue
https://storybook.grommet.io/?path=/story/visualizations-datatable-clickable--clickable-data-table

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by Utsav-Ladani**
#### What does this PR do?
* It fixes the space key interaction for clickable Datatable rows.

#### Where should the reviewer start?
* `src/js/components/DataTable/Body.js`

#### How should this be manually tested?
* Press `Space` key when row is selected.

#### What are the relevant issues?
* Closes #7335

#### Do the grommet docs need to be updated?
* No.

#### Should this PR be mentioned in the release notes?
* May be.

#### Is this change backwards compatible or is it a breaking change?
* It's backwards compatible.

### CODE REVIEW

**Review by britt6612 — COMMENTED** [review]
this looks good can we add a test for this functionality ?

File: src/js/components/DataTable/Body.js [hunk 1]
```diff
@@ -234,13 +234,20 @@ const Body = forwardRef(
...
-            }
-          }
-        }}
+        onSpace={
+          clickableRow
+            ? (event) => {
+                if (typeof onClickRow === 'function') {
+                  event.persist();
+                  const adjustedEvent = event;
+                  adjustedEvent.datum = data[active];
```
Comment (britt6612) [inline src/js/components/DataTable/Body.js:243]: maybe we can also add a check to make sure `data[active]` is defined

Comment (Utsav-Ladani) [reply src/js/components/DataTable/Body.js]: Done

**Review by britt6612 — APPROVED** [review]
LGTM

File: src/js/components/DataTable/Body.js [hunk 2]
```diff
@@ -234,13 +234,20 @@ const Body = forwardRef(
...
         }
         // The WCAG recommendation for checkboxes is to select them with "Space"
-        onSpace={() => {
-          if (clickableRow) {
-            if (onClickRow === 'select') {
-              selectRow();
-            }
-          }
-        }}
+        onSpace={
```
Comment (jcfilben) [inline src/js/components/DataTable/Body.js:237]: Can we prevent the page from scrolling when we interact with a row via the 'space' key?

**Review by jcfilben — COMMENTED** [review]
Nice work, just one comment

File: src/js/components/DataTable/Body.js (hunk 2 above)
Comment (Utsav-Ladani) [reply src/js/components/DataTable/Body.js:237]: @jcfilben Can you please screen recording of that behaviour?

Comment (jcfilben) [reply src/js/components/DataTable/Body.js:237]: Sure, here is a video

https://github.com/user-attachments/assets/d2d9ec3a-dd79-4c74-83d0-cc06024578e0

Comment (Utsav-Ladani) [reply src/js/components/DataTable/Body.js:237]: Done

**Comment by Utsav-Ladani** [comment]
@jcfilben I made changes and now pressing space key don't scroll the page.

**Review by MikeKingdom — APPROVED** [review]
LGTM

**Pull request merged**